### ISSUE TITLE
SafeRepr.test_brokenReprIncludesID

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6265]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6265|
|Type|enhancement|
|Created|2013-01-19 01:14:13Z|
|Branch|https://github.com/twisted/twisted/tree/improve-SafeRepr.test_brokenReprIncludesID-6265|

In [ticket:5544#18 [#5544](https://github.com/twisted/twisted/issues/5544)], exarkun suggests that assertion made by
`twisted.python.test.test_reflectpy3.SafeRepr.test_brokenReprIncludesID` is too weak.

It should probably assert about what exactly the string should look like, rather than just that it has a particular substring.

Attachments:

* [6265.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/f27/f27e4bad029e8c2a2bf8737bf12439133dd4b977/6163a4bd59cd0008424ae6ca3beba752e2541672.patch) (554 bytes) - added by hash14 on 2013-01-26 05:10:15Z - Stricter testing of id() in test_brokenReprIncludesID

<details><summary>Searchable metadata</summary>

```
trac-id__6265 6265
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_improve_SafeRepr_test_brokenReprIncludesID_6265 branches/improve-SafeRepr.test_brokenReprIncludesID-6265
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1358558053000000 1358558053000000
changetime__1360612093000000 1360612093000000
version__None None
owner__tom_prince tom.prince
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="hash14's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|hash14 commented|
|-|-|

Note that the patch assumes that className and formatter.__name__ in twisted.python._reflectpy3._safeFormat will always be BrokenType and repr respectively. Not sure if this is a safe assumption, please advise...

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @tomprince|
|-|-|

Thanks for your contribution.

When calling `safe_repr`, the `formatter` is going to be `repr` and `BrokenType` is a name coming from the test-suite, so in this test, it will always be that.

  1. There should be spaces around operators like `%`.
  1. A comment on why only the first line of the repr is looked at would be helpful for people looking at the code in the future.

I'll fix these two things up, and then merge.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince commented|
|-|-|

(In [37111]) Branching to improve-SafeRepr.test_brokenReprIncludesID-6265.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince commented|
|-|-|

(In [37112]) Apply patch from hash14.

Refs [#6265](https://github.com/twisted/twisted/issues/6265).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

Also, there was a twistedchecker error: String formatting operations should always use a tuple for non-mapping values

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince set status to `closed`|
|-|-|

(In [37116]) Merge improve-SafeRepr.test_brokenReprIncludesID-6265: Assert about the actual format of the repr.

Author: hash14
Reviewers: tom.prince
Fixes: [#6265](https://github.com/twisted/twisted/issues/6265)
